opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -326,6 +330,34 @@ namespace Opm
                         }
                         updatePrimaryVariables(simulator, well_state, deferred_logger);
                     }
+
+                    bool isNowGroupControl = ws.production_cmode == Well::ProducerCMode::GRUP || ws.injection_cmode == Well::InjectorCMode::GRUP;
+                    if (isGroupControl != isNowGroupControl) {
+                        // We need to update the globalIsGRUP vector and the group target reduction.
+                        // But we can not communicate it to the other ranks during the local iterations.
+                        // The globalIsGrup vector is thus only communicated after the local iterations are done.
+                        // This may give some discrepencies in non-linear behaviour between MPI runs,
+                        // but the solution should convergece to the same solution.
+                        well_state.updateGlobalIsGrup();
```

Comment:
Here, you do not need to update the whole `well_state`. I believe you should only update the info related to the current well, because other wells stay unchanged.